### ISSUE TITLE
Internationalization: custom disconnect message?

### ISSUE THREAD

**Issue by nicholaelaw** [rstudio/shiny#1828]
I asked this in the discussion group and got no answer yet. I couldn't find the origin in sources neither. 

Is it possible to change the notification after disconnecting? I mean this thing here:

![image](https://user-images.githubusercontent.com/10719482/29705200-9276ec10-89af-11e7-8e31-12885e71be4f.png)

This is essentially the first thing the user will see when returning to the app after a while of inactivity. Particularly so on mobile devices where the browser could not maintain connection when screen is turned off. 

**tl;dr**: I want to make this message 1) readable by people who don't know English (translation), and 2) a bit more friendly (custom HTML, perhaps. Pictures of cats. Anything other than a "this isn't working" screen.).

**Comment by jcheng5** [comment]
It's not currently customizable but that is something we should do. It would actually be a change to the server software (Shiny Server, Connect, and ShinyApps.io).

**Comment by nicholaelaw** [comment]
Well then should I submit a feature request in Shiny Server repo? Is this feature under consideration?

**Comment by tcash21** [comment]
Also agree this would be hugely helpful. Any updates or workarounds we can use in the meantime. Just want to show the user something a bit informative with a message on what to do next.

**Comment by michaelhogersnplm** [comment]
@tcash21 an alternative is to use the shiny:disconnected state, the below may require some extra tinkering. I recommend to use JavaScript to toggle the visibility of a modal for example, say ```$("#modalIdentifier").toggle()```. 

```
$(document).on('shiny:disconnected', function(event) {
// use JavaScript code here to show a customized pop-up
// ```$("#modalIdentifier").toggle()```
});
```

**Comment by daattali** [comment]
This isn't a perfect solution, but it's what I've been using for years. This is an example of what one of my apps looks like when disconnected:

![image](https://user-images.githubusercontent.com/952340/83962495-e8dd3680-a86b-11ea-84db-60f823789024.png)

I achieved it by adding the following CSS: (you can customize the text or the colours and style)

```
#ss-connect-dialog {
  position: absolute;
  top: 50px;
  bottom: auto;
  left: 50%;
  margin-left: -150px;
  font-size: 20px;
  width: 300px;
  background: #fff;
  color: #444;
  padding: 20px 30px;
  text-align: center;
}
#ss-connect-dialog::before {
  content: 'An error occurred. Please refresh the page and try again.';
}
#ss-connect-dialog label, #ss-connect-dialog a {
  display: none;
}
```

This isn't future-proof; if RStudio changes the HTML element that appears on disconnect, this may stop working. I hope they don't decide to do that :) Until then, feel free to use this method

**Comment by daattali** [comment]
For anyone who sees this in the future, I created a little package that gives you more control over the the disconnect message. You can change the text, colours, and other parameters.  See [`shinydisconnect package`](https://github.com/daattali/shinydisconnect). Example:

![image](https://user-images.githubusercontent.com/952340/83962495-e8dd3680-a86b-11ea-84db-60f823789024.png)

**Comment by tcash21** [comment]
Awesome, thanks so much Dean!

**Comment by dseynaev** [comment]
A package to workaround is nice, but should not be too hard to add in shiny itself?

Currently the greyout div seems to be hardcoded in JS:
https://github.com/rstudio/shiny/blob/4a2bb8fc4386ba1afa28b0cebee8c550707c18d6/inst/www/shared/shiny.js#L2608

The DOM element could be copied from an html template (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) which would be much easier to customize.

**Comment by chuansunaacg** [comment]
NVDA cannot read the text in the disconnect message. Any idea how to make it accessible to screen-readers (aria-label perhaps)? Also package solutions that I've tried (shinydisconnect, sever) are not accessible either.

**Comment by chuans01** [comment]
I have a follow-up question: is it possible to make the disconnect message accessible to screnn readers (NVDA etc.)? I can change the text but can't figure out how to make it read. I have also try packages that change the disconnect message (shinydisconnect, sever etc.) but neither is readable by a screen-reader.

Any help will be much appreciated!

Below is a minimal reproducible example:
`
library(shiny)
ui <- fluidPage(
  tags$head(tags$style(type="text/css",
                       "#shiny-disconnected-overlay:before {
    content: 'Testing. Your session has timed out.';
    text-align: center;
    font-size: 50pt;
    background-color: white;
    padding: 5%;
    width: 100%;
    position: absolute;
    top: 50%;
    transform: translateY(-50%);}")),
  actionButton("disconnect", "Disconnect the app")
)
server <- function(input, output, session) {
  observeEvent(input$disconnect, {
    session$close()
  })
}
shinyApp(ui, server)
`
